DataDog/ddprof
# Feature/leak detection

# What does this PR do?

Leak analysis tooling.
Also known as live heap. The aim is to highlight the memory that is still alive and relates to RSS usage.

# Motivation

Several requests to investigate leaks.

# Additional Notes

Some follow ups are expected to:
- Push the data to a different type of pprof
- Improve the documentation around the usage of the feature
- Input checks to prevent faulty usage of live mode option `mode=l`

# How to test the change?

If you compile the change, you can request it manually
```
ddprof -l debug --preset cpu_only -e "sALLOC mode=l" -u 30 ./test
```

You can also see a test here (private Datadog repo)
https://github.com/DataDog/prof-correctness/pull/38/files

Deploying this to staging, the overhead numbers seem reasonable.
